src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -444,18 +462,26 @@ internal static List<CompletionResult> CompleteModuleName(CompletionContext cont
                 }
             }
 
-            Exception exceptionThrown;
-            var psObjects = context.Helper.ExecuteCurrentPowerShell(out exceptionThrown);
+            var psObjects = context.Helper.ExecuteCurrentPowerShell(out _);
 
             if (psObjects != null)
             {
-                foreach (dynamic moduleInfo in psObjects)
+                foreach (PSObject item in psObjects)
                 {
-                    var completionText = moduleInfo.Name.ToString();
+                    var moduleInfo = (PSModuleInfo)item.BaseObject;
+                    var completionText = moduleInfo.Name;
                     var listItemText = completionText;
-                    var toolTip = "Description: " + moduleInfo.Description.ToString() + "\r\nModuleType: "
+                    if (shortNameSearch
+                        && completionText.Contains('.')
+                        && !shortNamePattern.IsMatch(completionText.Substring(completionText.LastIndexOf('.') + 1))
+                        && !shortNamePattern.IsMatch(completionText))
```

Comment:
```c#
                        && !shortNamePattern.IsMatch(completionText))
```
Why do we need the last check? From the PR description I see we should consider only last segment. Could you please show an example? Should we have a test for the scenario?

Reply:
It's so the user can type either the start of the module name as normal, or the short name at the end. Without it you wouldn't get `Microsoft.PowerShell.Archive` when typing in `Microsoft<Tab>`. As for whether or nor a test is needed, I don't think so because that check is to keep the old behavior and I assume there already are tests for tab completion of modules.